### PR TITLE
Implement `call` method for the `EVM module`

### PR DESCRIPTION
**Pull request by bkolad**
# Description
This PR introduces the following changes:
- `BlockEnv` & `EvmTransaction` types which are used in the `EVM` module.
-  Conversions between the aforementioned types and relevant `ReVM` types.
-  Implementation of `call` method for the `EVM` module
- `EVM` module tests

## Linked Issues
- Related to #413

## Testing
Added tests for the `EVM` module.

## Docs
The documentation will be provided after the module is fully implemented.

### CODE REVIEW

File: module-system/module-implementations/sov-evm/src/evm/conversions.rs [hunk 1]
```diff
@@ -25,3 +30,59 @@ impl From<ReVmAccountInfo> for AccountInfo {
...
+            // TODO: handle difficulty
+            difficulty: U256::ZERO,
+            prevrandao: block_env.prevrandao.map(|r| B256::from_slice(&r)),
+            basefee: U256::from_le_bytes(block_env.basefee),
+            gas_limit: U256::from_le_bytes(block_env.gas_limit),
+        }
+    }
+}
+
+impl From<AccessListItem> for (B160, Vec<U256>) {
```
Comment (citizen-stig) [inline module-system/module-implementations/sov-evm/src/evm/conversions.rs:49]: What does this tuple mean? Maybe we should create a regular struct with field names?

File: module-system/module-implementations/sov-evm/src/evm/test_helpers.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+use revm::primitives::{ExecutionResult, Output, B160};
+use std::path::PathBuf;
+
+pub(crate) fn output(result: ExecutionResult) -> Bytes {
+    match result {
+        ExecutionResult::Success { output, .. } => match output {
+            Output::Call(out) => out,
+            Output::Create(out, _) => out,
+        },
+        _ => panic!("Expected successful ExecutionResult"),
```
Comment (citizen-stig) [inline module-system/module-implementations/sov-evm/src/evm/test_helpers.rs:13]: Is it ok to panic here? Maybe use `TryFrom` ?

Comment (citizen-stig) [reply module-system/module-implementations/sov-evm/src/evm/test_helpers.rs:13]: oh, it is foreign type, but maybe we can return result

Comment (citizen-stig) [reply module-system/module-implementations/sov-evm/src/evm/test_helpers.rs:13]: It is for tests, nevermind

File: module-system/module-implementations/sov-evm/src/evm/tests.rs [hunk 3]
```diff
@@ -1,48 +1,25 @@
-use crate::{evm::AccountInfo, Evm};
```
Comment (citizen-stig) [inline module-system/module-implementations/sov-evm/src/evm/tests.rs:1]: What is the reason of it to be unit tests and not integration test?

Comment (citizen-stig) [reply module-system/module-implementations/sov-evm/src/evm/tests.rs:1]: I mean placing test outside of src and testing crate via public interface

File: module-system/module-implementations/sov-evm/src/evm/conversions.rs (hunk 1 above)
Comment (bkolad) [reply module-system/module-implementations/sov-evm/src/evm/conversions.rs:49]: This is why we have `AccessListItem`, but `revm` expects the above tuple (it is part of the interface and we can't change it) this is why we provide this conversions.

File: module-system/module-implementations/sov-evm/src/evm/test_helpers.rs (hunk 2 above)
Comment (bkolad) [reply module-system/module-implementations/sov-evm/src/evm/test_helpers.rs:13]: This is test helper and in testes I prefer to panic early.

File: module-system/module-implementations/sov-evm/src/evm/tests.rs (hunk 3 above)
Comment (bkolad) [reply module-system/module-implementations/sov-evm/src/evm/tests.rs:1]: The core logic of these tests relies on the public `EVM::call` method and could be separated into an integration test. However, to verify the success of the call, we are currently accessing the crate-private  `EVM` state. While we can make the appropriate query method public, it is currently tailored specifically for testing purposes. Instead, I suggest that we postpone changing the test to an integration test until we implement the `RPC` functionality for the `EVM` module. By doing so, we will be able to examine the internal state using the `RPC`, providing a more comprehensive testing approach.

Comment (citizen-stig) [reply module-system/module-implementations/sov-evm/src/evm/tests.rs:1]: Ok, as long as we plan to move to integration test eventually, I am good.

**Review by citizen-stig — APPROVED** [review]

**Pull request merged**